packages/table/src/common/grid.ts

```diff
@@ -253,7 +253,12 @@ export class Grid {
      * Returns the start and end indices of rows that intersect with the given
      * `Rect` argument.
      */
-    public getRowIndicesInRect(rect: Rect, includeGhostCells = false, limit = Grid.DEFAULT_MAX_ROWS): RowIndices {
+    public getRowIndicesInRect(
+        rect: Rect,
+        includeGhostCells = false,
+        rectHeightCorrection = 0,
```

Comment:
I think naming this `columHeaderHeight` would be simpler & more explicit

Reply:
Makes sense, there are some other functions using `getRowIndicesInRect` like `invokeOnVisibleCellsChangeCallback` that aren't using this arg but maybe they should be? since a row is technically not visible after it goes behind the header column but we still consider it visible since it's in the rect

Reply:
yeah, I guess `invokeOnVisibleCellsChangeCallback` should know about `columnHeaderHeight` as well...

if we're finding that the Grid needs to know the column header height so often, then maybe we should try to pass it `columnHeaderHeight` from Table / Table2 when we construct/validate the grid? hopefully this value won't get out of sync 🤞🏽, and that change might reduce some code complexity. before you do this, however, can you verify that `invokeOnVisibleCellsChangeCallback` is buggy right now? i.e. that it returns the wrong range of indices when the column header is larger than the default height.

Reply:
It's hard to test the accuracy of `invokeOnVisibleCellsChangeCallback` since its using a bleed buffer on both sides when ghost cells are enabled, but it does seem to be a bit off without accounting for the header height.
without accounting for header height:
![beforevisible](https://user-images.githubusercontent.com/13280642/163253659-ec7eef6e-faea-402c-8f5b-9c8a2d28bc0c.gif)

with accounting for header height:
![aftervisible](https://user-images.githubusercontent.com/13280642/163253652-fe10c683-6ce5-42c1-ae99-1ef6e9383d36.gif)

Reply:
It still seems off after accounting for header height... the "start" should be closer to 5 or 6 when you scroll all the way down, no? we can address this in a separate PR, but it looks like it's not a high priority for this particular callback since we haven't gotten any bug reports for `onVisibleCellsChange` accuracy